src/editor/components/flow/new-node-dialog.tsx

```diff
@@ -0,0 +1,127 @@
+import { Dialog } from '../dialog'
+import { Button } from '../form/button'
+import { Node as ReactFlowNode } from 'react-flow-renderer/nocss'
+
+interface AddNodeDialogProps {
+  isAddNodeOpen: boolean,
+
+  setIsAddNodeOpen( value: boolean ): void
+
+  save(): void
+
+  flow: any
+
+  nodes: any
+
+  setNodes( value: any ): void
+}
+
+export const AddNodeDialog = ( {
+  isAddNodeOpen,
+  setIsAddNodeOpen,
+  save,
+  flow,
+  nodes,
+  setNodes
+}: AddNodeDialogProps ): JSX.Element => {
+
+  const randomIntFromInterval = ( min: number, max: number ): number => {
+    return Math.floor( Math.random() * ( max - min + 1 ) + min )
+  }
+
+  const newHttpInNode = (): ReactFlowNode => {
```

Comment:
Can we make this less repetitive? Maybe by using a class for each node so you can just use 'new HttpInputNode()'

Reply:
yea I thought about that. I'll make some classes to clear up the new-node-dialog component